### PR TITLE
Made data sent to jsp minimal and refactor ProfileServletTest.java

### PR DESCRIPTION
**Pull request by ElleTojaroon**

### CODE REVIEW

**Comment by ElleTojaroon** [comment]
In the first commit, I also fixed the timezone abbreviation for #47

File: src/main/webapp/WEB-INF/view/profile.jsp [hunk 1]
```diff
@@ -119,15 +119,11 @@ User owner = (User) request.getAttribute("owner");
...
-                                DateTimeFormatter.ofPattern("EEE MMM dd HH:mm:ss zzz yyyy")
-                                    .withZone(ZoneId.systemDefault())
-                                    .format(message.getCreationTime());
+                        for (String displayTime : messageDisplayTimeToMessageContent.keySet()) {
                     %>
                         <li>
-                            <strong><%= creationTime %>:</strong>
-                            <%= message.getContent() %></li>
+                            <strong><%= displayTime %>:</strong>
+                            <%= messageDisplayTimeToMessageContent.get(displayTime) %></li>
```
Comment (krtejeda) [inline src/main/webapp/WEB-INF/view/profile.jsp]: I'm unfamiliar with ImmutableMaps, but do they somehow handle if two messages have the same creation time?

Comment (ElleTojaroon) [reply src/main/webapp/WEB-INF/view/profile.jsp]: In ProfileServlet.getMessageDisplayTimeToMessageContentOfUser(), I put merger function that handles the case when a duplicate key gets put. I, however, simplify the construction process, but still have a merger function in the stream processing.

**Comment by ElleTojaroon** [comment]
@krtejeda I uploaded changes and replied to your comments. PTAL.

**Pull request merged**